fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexMaintainer.java

```diff
@@ -301,4 +301,12 @@ public RecordCursor<FDBIndexedRawRecord> scanRemoteFetch(@Nonnull final IndexSca
         // Not implemented by default - needs to be overridden by individual maintainers
         throw new UnsupportedRemoteFetchIndexException("scanRemoteFetch operation is not supported by this index maintainer for Index " + state.index.getName());
     }
+
+    /**
+     * If the index is merge'Able, perform an index merge (within a single transaction limit).
+     * @return a future that completes the merge operation
+     */
+    @API(API.Status.EXPERIMENTAL)
+    public abstract CompletableFuture<Void> mergeIndex();
```

Comment:
This can probably just be an index operation, thereby admitting that it's very specific to the kind of index, which I think it is.